### PR TITLE
move database to class extension

### PR DESCRIPTION
**Pull request by bsouthga**
Add `TemporalDatabase` as a decendent of `promised-mongo`'s `Database`.

(also adds aggregate api fix to emulate promised-mongo's approach: https://github.com/gordonmleigh/promised-mongo/blob/master/lib/Collection.js#L40-L43

### CODE REVIEW

**Comment by yangchristian** [comment]
Great! This looks like a nice, low risk refactor. Merging now :+1:

File: lib/classes/TemporalCollection.js [hunk 1]
```diff
@@ -90,12 +90,46 @@ export default class TemporalCollection extends PromisedMongoCollection {
...
+      pipeline.unshift({ $match: restriction({}, date) });
+    }
+
+    return this.aggregateRaw(...pipeline);
   }
 
-  aggregateByDate(pipeline={}, options, date) {
-    return this.aggregateRaw({ ...pipeline, $match: addDate(pipeline.$match, date) }, options);
+  aggregate(...pipeline) {
+    return this._aggregateWrapper(addCurrent, null, pipeline);
```
Comment (yangchristian) [inline lib/classes/TemporalCollection.js]: I'm a little confused by the spread operators. Shouldn't the first argument to `aggregate` be an array already?

While it's true that `promised-mongo` passes an object in the example you linked (and in their [`aggregate` test](https://github.com/gordonmleigh/promised-mongo/blob/master/test/Collection.js#L28), I think this is actually a bug and should be passing an array (which is the Mongo API):
https://github.com/gordonmleigh/promised-mongo/issues/27

Our app code uses that array format too.

File: test/tpmongo_test.js [hunk 2]
```diff
@@ -374,7 +374,7 @@ describe('tpmongo', function() {
       return db.tempCollection.update({a: 1}, {$set: {b: 4}}, {multi: true});
     })
     .then(function() {
-      return db.tempCollection.aggregate({}, {$group: {_id: '$a', total: {$sum: '$b'}}});
+      return db.tempCollection.aggregate({$group: {_id: '$a', total: {$sum: '$b'}}});
```
Comment (yangchristian) [inline test/tpmongo_test.js]: Per comment above, this test fails if I change the call to use array format:
```js
return db.tempCollection.aggregate([{$group: {_id: '$a', total: {$sum: '$b'}}}]);
```

**Comment by bsouthga** [comment]
updated to require array pipeline

**Pull request merged**